operator/internal/manifests/storage/schema.go

```diff
@@ -0,0 +1,49 @@
+package storage
+
+import (
+	"time"
+
+	lokiv1beta1 "github.com/grafana/loki/operator/api/v1beta1"
+)
+
+// UpdateSchemas returns a list of schemas that can be used by Loki. It will append a new
+// schema if there are no existing, the last schema is not the latest, or the last schema
+// is not using the right object storage.
+func UpdateSchemas(version lokiv1beta1.ObjectStorageSchemaVersion, existing []Schema) []Schema {
+	count := len(existing)
+	date := time.Now().UTC()
+
+	if count == 0 {
+		return []Schema{
+			{
+				From:    date.Format("2006-01-02"),
+				Version: version,
+			},
+		}
+	}
+
+	last := existing[count-1]
+
+	if last.Version != version {
+		// Apply the changes to the index for logs starting the next day.
+		// Since the timestamps work by UTC, the goal is to avoid the scenario
+		// in which logs might be inaccessible because they were created with the
+		// wrong schema.
+		from := date.Add(time.Hour * 24).Format("2006-01-02")
```

Comment:
I am wondering if we should have a check for a grace period of a couple of hours (or however long we think it is gonna take to roll-out the changes in the worst case to all components). 

With this code let's say you would update the SchemaVersion at 1 second to UTC midnight, you would get the new schema 1 second later, but the change hasn't been rolled out to all components.

Reply:
Maybe it might be best to do a two day gap instead of a one day one.

Reply:
I do have a question about multiple schemas being applied for the same day. For instance, if I make a schema change to be applied on a day in the future and then make another to apply on the same day, is the most recent change the only one that matters?

Reply:
Yes there can only be a single schema that is valid for the full UTC day. 

I think we should make sure we do never change a schema that is active or will become active in the next n hours (e.g. 24 hours).